core/src/main/java/contacts/core/log/LoggerRegistry.kt

```diff
@@ -0,0 +1,30 @@
+package contacts.core.log
+
+import contacts.core.CrudApi
+import contacts.core.Redactable
+import contacts.core.redactedCopyOrThis
+
+class LoggerRegistry @JvmOverloads constructor(
+   private val logger: Logger = EmptyLogger(),
+) {
+
+    internal val apiListener: CrudApi.Listener = Listener(logger)
+
+    // Prevent consumers from invoking the listener functions by not having the registry implement
+    // it directly.
+    private inner class Listener(
+        private val logger: Logger,
```

Comment:
This `logger` actually overshadows the outer `logger`. I think I know what you are trying to do here. This whole thing can be simplified to,

```kotlin
class LoggerRegistry(logger: Logger = EmptyLogger()) {

    internal val apiListener: CrudApi.Listener = Listener(logger)

    // Prevent consumers from invoking the listener functions by not having the registry implement
    // it directly.
    private class Listener(private val logger: Logger) : CrudApi.Listener {
        
        override fun onPreExecute(api: CrudApi) {
            logRedactable(api)
        }

        override fun onPostExecute(result: CrudApi.Result) {
            logRedactable(result)
        }

        private fun logRedactable(redactable: Redactable) {
            logger.log(redactable.redactedCopyOrThis(logger.redactMessages).toString())
        }
    }
}
```

Reply:
I made the update here https://github.com/vestrel00/contacts-android/commit/e321874e5c37739a6cb0a26ac479400a662b911e